javaharness/java/arcs/android/impl/ArcsLocal.java

```diff
@@ -34,10 +34,16 @@
 
   @Override
   public void runArc(ArcData arcData) {
-    if (arcData.getParticle() != null) {
-      PECInnerPort pecInnerPort =
-          pecPortManager.getOrCreateInnerPort(arcData.getPecId(), arcData.getSessionId());
-      pecInnerPort.mapParticle(arcData.getParticle());
+    PECInnerPort pecInnerPort = null;
+    for (int i = 0; i < arcData.getParticleList().size(); ++i) {
```

Comment:
for (ArcData.ParticleData particleData : arcData.getParticleList()) { ... }

Reply:
done